### ISSUE TITLE
RSeqQC: parse Transcript Integrity Number

### ISSUE THREAD

**Issue by drbecavin** [ewels/MultiQC#737]
Integration of Transcript Integrity Number from RSeqQC

* **Name of tool:**
    * Transcript Integrity Number - `tin.py`
* **Tool description:**
    * This program is designed to evaluate RNA integrity at transcript level. TIN (transcript integrity number) is named in analogous to RIN (RNA integrity number). RIN (RNA integrity number) is the most widely used metric to evaluate RNA integrity at sample (or transcriptome) level. It is a very useful preventive measure to ensure good RNA quality and robust, reproducible RNA sequencing.
* **Tool homepage:**
    * http://rseqc.sourceforge.net/#tin-py
* **Complete log file output:**
    * [New_abx_ZT13_IP_4.summary.txt](https://github.com/ewels/MultiQC/files/1931254/New_abx_ZT13_IP_4.summary.txt)
* **Log filename pattern:**
    * `.summary.txt`  and `.tin.xls`
* **Most interesting data for General Stats table:**
    * TIN(median)
* **Data suitable for MultiQC plot(s):**
    * TIN(median)

Thanks for your amazing job on MultiQC. I love this softwareeeeee ! It changed my life !

**Comment by ewels** [comment]
Hah, thanks for the comment @drbecavin - I should add that quote to the website testimonials 😉 

Should be pretty easy to add this. I'll take a look into it when I get a chance.

**Comment by drbecavin** [comment]
No problem, I really enjoy the way you manage all these logs, and the quality of the html report created. Thanks!
Maybe I should create another issue for read_quality.py, another tool of RSeQC?

**Comment by ewels** [comment]
Sure 👍 It's good to have multiple issues where possible to break things up.

**Comment by EngineerReversed** [comment]
> Integration of Transcript Integrity Number from RSeqQC
> 
> * **Name of tool:**
>   
>   * Transcript Integrity Number - `tin.py`
> * **Tool description:**
>   
>   * This program is designed to evaluate RNA integrity at transcript level. TIN (transcript integrity number) is named in analogous to RIN (RNA integrity number). RIN (RNA integrity number) is the most widely used metric to evaluate RNA integrity at sample (or transcriptome) level. It is a very useful preventive measure to ensure good RNA quality and robust, reproducible RNA sequencing.
> * **Tool homepage:**
>   
>   * http://rseqc.sourceforge.net/#tin-py
> * **Complete log file output:**
>   
>   * [New_abx_ZT13_IP_4.summary.txt](https://github.com/ewels/MultiQC/files/1931254/New_abx_ZT13_IP_4.summary.txt)
> * **Log filename pattern:**
>   
>   * `.summary.txt`  and `.tin.xls`
> * **Most interesting data for General Stats table:**
>   
>   * TIN(median)
> * **Data suitable for MultiQC plot(s):**
>   
>   * TIN(median)
> 
> Thanks for your amazing job on MultiQC. I love this softwareeeeee ! It changed my life !

Has this been included in `MultiQC_report`?

**Comment by guidohooiveld** [comment]
Hi, I also kindly second this request to include the results of `tin.py` in a MultiQC report.
An example picture (box plot; Figure 1B) of representing the data of multiple samples can be found in [this paper](https://www.hindawi.com/journals/bmri/2018/2906292/fig1/), but other representations of the results (median TIN score + SD or IQR of all transcripts in a sample) in a table or graph may be more appropriate... ??

For completeness below the code used to run the TIN module, and the results have been attached. (the txt file `*out.summary.txt` contains the summary of the sample (i.e. median + SD), the xls [actually, also a tab delim txt] file `*out.tin.xls` the TIN score per transcript). 

```
[guidoh@localhost P15-1-6h]$ tin.py -i P15-1-6h_Aligned.sortedByCoord.out.bam \
-r /mnt/files/guido/INDEX/STAR/Housekeeping_TranscriptsHuman2158.bed
@ 2021-06-10 11:44:04: Get BAM file(s) ...
Total 1 BAM file(s):
        P15-1-6h_Aligned.sortedByCoord.out.bam
@ 2021-06-10 11:44:04: Processing P15-1-6h_Aligned.sortedByCoord.out.bam
 [guidoh@localhost P15-1-6h]$ 

```
Thank you for having a look at this!
G

[output_tin.py.zip](https://github.com/ewels/MultiQC/files/6632165/output_tin.py.zip)

**Comment by ewels** [comment]
Hi all,

Apologies for the very (very, very) long time it's taken to get this added. @ErikDanielsson has just put together a new RSeQC submodule to support this output in #1481 and it will be part of the v1.11 release any day now.

In the end, I decided that we should keep it simple. It adds two columns to the _General Statistics_ table: the median TIN and the stdev. The latter is hidden by default, it can be shown via the _Configure Columns_ button or at report generation time via a config (see [docs](https://multiqc.info/docs/#hiding-columns)).

I hope this is still helpful to you all, despite coming over 3 years late! 😁 Shout if you hit any problems with it.

Many thanks,

Phil

**Comment by guidohooiveld** [comment]
Thanks Phil and Erik for creating the MultiQC RSeQC TIN submodule; much appreciated!

I updated MutiQC to the latest development version, and ran it again on a map containing various QC output files, including TIN.
Et voila, the 2 columns (of which one is hidden) were indeed added to the General Statistics table. Nice & thanks!



One comment/question, though, regarding the sample names used for the TIN values in the General Statistics table: these are not the same as used for the other RSeQC modules. This makes the table 'less nice' and more difficult to read. See screenshot below.

I think this is due to the fact that within the TIN "summary" file (the txt file `*out.summary.txt`) the full name of the BAM file is returned (used) by RSeQC, which is then extracted (parsed) by the MultiQC TIN module, and subsequently used in the General Statistics table.

Therefore: would you have any suggestion to prevent this form happening? So that only the 'base name' is used in the table? Maybe by somehow using on-the-fly the function `fn_clean_sample_names`? But I am not an expert on this and it may be a too naive thought...?


Thus, in summary: in the General Statistics table the full name present in the TIN  summary file (`*out.summary.txt`) is used (e.g. "P26-1-6h_Aligned.sortedByCoord.out.bam"), whereas just the use of only the sample ID (base name) "P26-1-6h" would be preferred.

Content TIN summary file (`P26-1-6h_Aligned.sortedByCoord.out.summary.txt`):
```
Bam_file	TIN(mean)	TIN(median)	TIN(stdev)
P26-1-6h_Aligned.sortedByCoord.out.bam	53.72327495737302	53.34221052273402	18.530355596890026

```
An example file is present in my previous post in this thread (https://github.com/ewels/MultiQC/issues/737#issuecomment-858673304).
	 

![image](https://user-images.githubusercontent.com/22979991/124501559-158eb500-ddc2-11eb-9646-c38fadb16985.png)

**Comment by ewels** [comment]
Ah yes, our first v1.11 release bug! You're totally right, we missed passing the sample name through the `self.clean_s_name()` function ([docs](https://multiqc.info/docs/#custom-sample-names)). It's a one-line fix, I'll try to get to it later today.

**Comment by ewels** [comment]
Moved into a dedicated issue: https://github.com/ewels/MultiQC/issues/1484

**Comment by ewels** [comment]
(fixed in v1.12dev)
